core/templates/dev/head/components/state-editor/state-responses-editor/state-responses.directive.ts

```diff
@@ -568,6 +568,47 @@ angular.module('oppia').directive('stateResponses', [
             var activeStateName = $scope.stateName;
             return outcome && (outcome.dest === activeStateName);
           };
+
+          var getDeletedChoicesIndex = function(
+              newAnswerChoices, oldAnswerChoices) {
+            oldAnswerChoicesArray = oldAnswerChoices.map(function(elem) {
+              return elem.label;
+            }) ;
+            newAnswerChoicesArray = newAnswerChoices.map(function(elem) {
+              return elem.label;
+            }) ;
+            var deletedChoicesIndex = oldAnswerChoicesArray.filter(
+              function(elem) {
+                return newAnswerChoicesArray.indexOf(elem) === -1;
+              }
+            ).map(function(elem) {
+              return oldAnswerChoicesArray.indexOf(elem) ;
+            });
+            var finalDeletedChoicesIndex = [];
+            for (var i = 0; i < deletedChoicesIndex.length; i++) {
+              finalDeletedChoicesIndex.push(deletedChoicesIndex[i] - i);
+            }
+            return finalDeletedChoicesIndex;
+          };
+
+          var validateAnwerGroupsOnChoiceDeletion = function(index) {
+            var requiresUpdation = false;
+            var answerGroups = ResponsesService.getAnswerGroups();
+            for (var i = 0; i < answerGroups.length; i++) {
+              var rules = answerGroups[i].rules;
+              for (var j = 0; j < rules.length; j++) {
+                if (index < rules[j].inputs.x) {
```

Comment:
`index` is the index of the deleted option, right? I can understand the make invalid section below, but why is there a 'lesser than' comparison here?
Also, just to confirm `rules[j].inputs.x` is the index of the choice for that rule, right?

Reply:
> index is the index of the deleted option, right?

Yes.

> , but why is there a 'lesser than' comparison here?

When a choice having index less than the `rules[j].inputs.x`(answer), We need to shift the rule index up by one position in order to point to the same choice.For example, The rule index is 4 and we delete a choice with index 2, we need to make the rule idex point to the 3rd choice(Which was previously the 4th choice before deletion).

> just to confirm rules[j].inputs.x is the index of the choice for that rule

Yes., this refers to the index of the answer choice.

Reply:
Ah, ok, thanks for the explanation!